proxy/proxy.js

```diff
@@ -74,6 +84,7 @@ const authHard = (req, res, next) => {
 
 // Setup all the proxy routes.
 
+app.use('/legacy_api', authSoft);
```

Comment:
for some reason github doesn't let me add comments anywhere in the file

can you add another TODO - eventually we'll want to remove `authSoft` for requests to `/api`

Reply:
Why will we remove authSoft?

Reply:
Right now the API still looks at the header info for auth information. The flow is currently this: incoming request -> request /api/auth/check -> set headers -> proxy request along to /api/whatever

But the API can just inspect the cookie and figure out auth info - it doesn't really need the x-coursetable-* headers anymore. Once we make that change, we won't need to call authSoft for /api

Reply:
Makes sense, that's really interesting